src/LogicTasks/Syntax/TreeToFormula.hs

```diff
@@ -74,11 +75,16 @@ partialGrade _ sol
 
 
 
-completeGrade :: OutputMonad m => SynTreeInst -> TreeFormulaAnswer -> LangM m
-completeGrade inst sol
-    | fromJust ( maybeTree sol) /= tree inst = reject $ do
-      english "Your solution is not correct."
-      german "Ihre Abgabe ist nicht die korrekte Lösung."
+completeGrade :: (OutputMonad m, MonadIO m) => FilePath -> SynTreeInst -> TreeFormulaAnswer -> LangM m
+completeGrade path inst sol
+    | fromJust ( maybeTree sol) /= tree inst = refuse $ do
+        instruct $ do
+          english "Your solution is not correct. The syntax tree for the entered formula looks like this:"
+          german "Ihre Abgabe ist nicht die korrekte Lösung. Der Syntaxbaum zu der eingegebenen Formel sieht so aus:"
+
+        image $=<< liftIO $ cacheTree (transferToPicture (fromJust (maybeTree sol))) path
```

Comment:
The two occurrences of `fromJust (maybeTree sol)` could be shared by something like
```haskell
  where
    treeAnswer = fromJust (maybeTree sol)
```